Add the Tekton Documentation Contributor's Guide and Style Guide.

sergetron opened the pull request:
Adds the Tekton Documentation Contributor's Guide and Style Guide.

This PR just adds the relevant files. I will link to those files from CONTRIBUTE.md and other places in the Tekton docs in a separate PR.

AlanGreene commented:
General question: do we have a preference for active over passive voice in the documentation?

sergetron commented:
We should always use active. It's easier to parse and more user-friendly. I
think I mentioned it in the "Writing High Quality Documentation" doc and I
can also reiterate it in the "Content Guidelines" doc.

On Thu, Apr 30, 2020 at 10:41 AM Alan Greene <notifications@github.com>
wrote:

> General question: do we have a preference for active over passive voice in
> the documentation?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tektoncd/website/pull/100#issuecomment-621896834>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG3UR4XOOLNN4ZP6NMPCL3DRPGEZPANCNFSM4MUAL3BQ>
> .
>

AlanGreene commented:
I didn't see it called out explicitly so may have missed it. It's definitely included in the Google style guide under 'Language and grammar' (and in the [highlights](https://developers.google.com/style/highlights)). If we're pointing at the Google style guide my only concern is that it could be quite intimidating for newcomers if they feel they have to digest all of that before contributing docs.

So far the content is looking great :+1: I'll give it a proper read through this evening and add any other feedback.

sergetron commented:
Just making sure, did you read all four files in the PR? The idea is we
cover all the most important things in our guide. The link to the
Google style guide is there if they want to go beyond what our guide covers.

On Thu, Apr 30, 2020 at 10:53 AM Alan Greene <notifications@github.com>
wrote:

> I didn't see it called out explicitly so may have missed it. It's
> definitely included in the Google style guide under 'Language and grammar'
> (and in the highlights <https://developers.google.com/style/highlights>).
> If we're pointing at the Google style guide my only concern is that it
> could be quite intimidating for newcomers if they feel they have to digest
> all of that before contributing docs.
>
> So far the content is looking great 👍 I'll give it a proper read through
> this evening and add any other feedback.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tektoncd/website/pull/100#issuecomment-621905024>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG3UR4UE26JNQPEKQRYREWLRPGGIDANCNFSM4MUAL3BQ>
> .
>

AlanGreene commented on content/en/doc-con-formatting.md:
```diff
@@ -0,0 +1,158 @@
+    </tr>
+  </table>
+
+- Use angle brackets for placeholders, such as variables, and tell the reader what the placeholders represents. For example:
```
```suggestion
- Use angle brackets for placeholders, such as variables, and tell the reader what the placeholders represent. For example:

```

sergetron replied:
Fixed.

AlanGreene commented on content/en/doc-con-main.md:
```diff
@@ -0,0 +1,99 @@
+[Tekton repository](https://github.com/tektoncd). Documentation for each project is stored
+in the `docs` directory within the repository. 
+
+Tekton documentation is mirrored on the [Tekton website](https://www.tekton.dev) with the GitHub
```
```suggestion
Tekton documentation is mirrored on the [Tekton website](https://tekton.dev) with the GitHub
```

the canonical URL omits `www.`

sergetron replied:
Didn't know that! Fixed.

AlanGreene commented on content/en/doc-con-writing.md:
```diff
@@ -0,0 +1,84 @@
+accomplish each task?
+
+Write down headings representing each of those major points or areas. For example, “Installation,”
+“Configuration,” “Operation modes,” or Command-line options.” These major things become the outline
```
```suggestion
“Configuration,” “Operation modes,” or “Command-line options.” These major things become the outline

```

sergetron replied:
Fixed.

AlanGreene commented on content/en/doc-con-content.md:
```diff
@@ -0,0 +1,171 @@
+
+## Style and tone
+
+- **Never use "should" or "might." **These words introduce uncertainty. We're either sure how our software works or we're not.
```
```suggestion
- **Never use "should" or "might."** These words introduce uncertainty. We're either sure how our software works or we're not.
```

sergetron replied:
Fixed.

AlanGreene commented on content/en/doc-con-content.md:
```diff
@@ -0,0 +1,171 @@
+    </tr>
+  </table>
+
+- **Never use "might" or "may." **"May" is often confused with "might" but they are not the same. "Might" introduces uncertainty just like "should," while "may" implies the permission has been granted and not that an option or action is available to the user. For example:
```
```suggestion
- **Never use "might" or "may."** "May" is often confused with "might" but they are not the same. "Might" introduces uncertainty just like "should," while "may" implies the permission has been granted and not that an option or action is available to the user. For example:
```

sergetron replied:
Fixed.

AlanGreene commented on content/en/doc-con-content.md:
```diff
@@ -0,0 +1,171 @@
+## Things to avoid
+
+Avoid the following:
+- **Copy-pasting large portions of existing documentation.** If you need to reuse a large piece of content,
```
```suggestion

- **Copy-pasting large portions of existing documentation.** If you need to reuse a large piece of content,
```
need a blank line before start of a list to ensure proper formatting

sergetron replied:
Fixed.

AlanGreene commented on content/en/doc-con-formatting.md:
```diff
@@ -0,0 +1,158 @@
+
+## User interface elements
+
+- Use **bold **to indicate anything that's clickable, such as buttons or menu items.
```
```suggestion
- Use **bold** to indicate anything that's clickable, such as buttons or menu items.
```

sergetron replied:
Fixed.

AlanGreene commented on content/en/doc-con-formatting.md:
```diff
@@ -0,0 +1,158 @@
+
+## Links
+
+- **Never use "here" or "this" as a link anchor. **Use either the exact title of the document being
```
```suggestion
- **Never use "here" or "this" as a link anchor.** Use either the exact title of the document being
```

sergetron replied:
Fixed.

sergetron commented:
Thank you for the thorough review Alan! I added an explicit bullet about active vs. passive voice and fixed all the formatting issues. I'll be adding the new commits shortly.

AlanGreene commented:
`sergetron@` should be `@sergetron` but let's get this in now and we can fix it up after since it's not actually linked from anywhere yet. I don't think GitHub usernames are automatically clickable on the website anyway, only when the markup is viewed in the GitHub UI.

/approve
/meow
